### PR TITLE
Fix warning on Win64 build

### PR DESCRIPTION
**Pull request by aafemt**
Also according to MSDN _beginthreadex() return error code in errno, not LastError.

### CODE REVIEW

File: src/common/ThreadStart.cpp [hunk 1]
```diff
@@ -309,13 +309,12 @@ ThreadId Thread::start(ThreadEntryPoint* routine, void* arg, int priority_arg, H
...
-	unsigned long real_handle =
-		_beginthreadex(NULL, 0, THREAD_ENTRYPOINT, THREAD_ARG, CREATE_SUSPENDED, &thread_id);
-	if (!real_handle)
+	HANDLE handle =
+		reinterpret_cast<HANDLE>(_beginthreadex(NULL, 0, THREAD_ENTRYPOINT, THREAD_ARG, CREATE_SUSPENDED, &thread_id));
+	if (!handle)
 	{
-		Firebird::system_call_failed::raise("_beginthreadex", GetLastError());
+		Firebird::system_call_failed::raise("_beginthreadex", errno);
 	}
```
Comment (dyemanov) [inline src/common/ThreadStart.cpp]: IIRC, system_call_failed / system_error exceptions expect Win32 error codes rather then CRT ones.

**Comment by aafemt** [comment]
> IIRC, system_call_failed / system_error exceptions expect Win32 error codes rather then 
> CRT ones.

errno is used with system_call_failed everywhere on POSIX and in a couple of places on 
Win32 so I don't see difference.

**Comment by AlexPeshkoff** [comment]
On POSIX use of errno in system_call_failed is as designed, on win32 - sooner of all a couple of bugs

**Comment by aafemt** [comment]
Which exception class should be used here then?
As I said, 
https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/beginthread-beginthreadex?view=vs-2019 
clearly states that error is returned in errno.

**Comment by dyemanov** [comment]
> clearly states that error is returned in errno
Because it's a CRT function, they are somewhat isolated from the system calls. But it doesn't necessarily mean that SetLastError() is not called under the hood. Google shows a number of recommendations to check GetLastError() after failing _beginthreadex(), so it might actually work. So I'd revert this your change back (or replace errno with ERRNO which is defined as GetLastError() on Windows).

**Comment by aafemt** [comment]
In current versions of CRT it works because no other system functions are called after 
CreateThread() as you can see in ucrt sources but they can change it any time.

File: src/common/ThreadStart.cpp (hunk 1 above)
Comment (aafemt) [reply src/common/ThreadStart.cpp]: Ok, I've reverted to GetLastError() with comment.

**Pull request merged**